packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1835,17 +1835,24 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
       errorMaxLines: decoration.errorMaxLines,
     );
 
-    final Widget counter = decoration.counterText == null ? null :
-      Semantics(
-        container: true,
-        liveRegion: isFocused,
-        child: Text(
-          decoration.counterText,
-          style: _getHelperStyle(themeData).merge(decoration.counterStyle),
-          overflow: TextOverflow.ellipsis,
-          semanticsLabel: decoration.semanticCounterText,
-        ),
-      );
+    Widget counter;
+    if (decoration.counterText == '' && decoration.counter == null) {
+      counter = null;
+    } else if (decoration.counter != null) {
+      counter = decoration.counter;
```

Comment:
Would it be possible for someone creating a custom counter widget to provide the same accessibility guarantees that the current widget does? For example, can you find out about the focus state and so on.

If users provide their own counters, is it obvious how to do so?

Reply:
Good call.  I just added `isFocused` to the function parameters so that users can support this.  I also called this out in the docs and added an example of how to do it right.